openedx/core/djangoapps/user_authn/views/tests/test_logout.py

```diff
@@ -61,16 +62,19 @@ def _authenticate_with_oauth(self, oauth_client):
         self.assertListEqual(self.client.session[AUTHORIZED_CLIENTS_SESSION_KEY], [oauth_client.client_id])
 
     @ddt.data(
-        ('/courses', 'testserver'),
-        ('https://edx.org/courses', 'edx.org'),
-        ('https://test.edx.org/courses', 'edx.org'),
+        (True, '/courses', 'testserver'),
+        (True, 'https://edx.org/courses', 'edx.org'),
+        (True, 'https://test.edx.org/courses', 'edx.org'),
+        (True, '/courses/course-v1:ARTS+D1+2018_T/course/', 'edx.org'),
+        (False, '/courses/course-v1:ARTS+D1+2018_T/course/', 'edx.org'),
+        (False, '/enterprise/c5dad9a7-741c-4841-868f-850aca3ff848/course/Microsoft+DAT206x/enroll/', 'edx.org'),
```

Comment:
Can we add more test cases with query strings having plus and space?

Reply:
I have added 2 more test with redirect URL containing query param.
1. `www.example.com/logout?next=/courses/?q=computer+science` (only query param of redirect URL is encoded )